server/aesdsocket.c

```diff
@@ -117,36 +248,242 @@ void print_closed_conn(struct sockaddr_storage client_addr)
+                            }
+
+                            continue;
+                        }
+
+                        //Else, error occurred, print it to syslog and finish program
+                        syslog(LOG_ERR, "Could not read from the file: %s", strerror(errno));
+                        ret = pthread_mutex_unlock(thread_info_parsed->mutex);
+                        if(ret != 0)
+                        {
+                            syslog(LOG_ERR, "Could not unlock mutex: %s", strerror(ret));
+                        }
+                        goto exit_all; 
+                    }
+
+                    //Less bytes remaining
+                    to_be_sent -= read_bytes;
+                    
+                    //Send the contents back to the client
+                    int sent_bytes = -1;
+                    int send_off = 0;
+                    while(sent_bytes != 0)
+                    {
+                        sent_bytes = send(thread_info_parsed->socket_client, &buff_read[send_off], send_bytes, 0);
+                        if(sent_bytes == -1)
+                        {
+                            //If the error is caused by an interruption of the system call try again
+                            if(errno == EINTR)
+                            {
+                                //Free the mutex
+                                ret = pthread_mutex_unlock(thread_info_parsed->mutex);
+                                if(ret != 0)
+                                {
+                                    syslog(LOG_ERR, "Could not unlock mutex: %s", strerror(ret));
+                                }
+                                continue;
+                            }
+
+                            //Else, error occurred, print it to syslog and finish program
+                            syslog(LOG_ERR, "Could not read from the file: %s", strerror(errno));
+                            ret = pthread_mutex_unlock(thread_info_parsed->mutex);
+                            if(ret != 0)
+                            {
+                                syslog(LOG_ERR, "Could not unlock mutex: %s", strerror(ret));
+                            }
+                            goto exit_all; 
+                        }
+                        send_bytes -= sent_bytes;
+                        send_off += sent_bytes;
+                    }
+                    
+                }
+
+                //Reset index to use the malloc'ed buffer from the beginning
+                index = 0;
+
+                //The lock can be freed
+                ret = pthread_mutex_unlock(thread_info_parsed->mutex);
+                if(ret != 0)
+                {
+                    syslog(LOG_ERR, "Could not unlock mutex: %s", strerror(ret));
+                    goto exit_all; 
+                }
+            }
+            //Realloc the array if it got full without an '\n'
+            else if(index == (RECV_BUFF_LEN*chunks))
```

Comment:
@jordicrosmompart I think you'll still have the issue from https://github.com/cu-ecen-aeld/assignments-3-and-later-jordicrosmompart/pull/3#discussion_r988337086 here.